app/lib/features/events/widgets/change_date_sheet.dart

```diff
@@ -0,0 +1,356 @@
+                        hintText: L10n.of(context).selectDate,
+                        suffixIcon: const Icon(Icons.calendar_month_outlined),
+                      ),
+                      onTap: () => _selectDate(isStartDate: false),
+                      validator: (value) {
+                        if (value != null && value.isEmpty) {
+                          return L10n.of(context).endDateRequired;
+                        }
+                        return null;
+                      },
+                    ),
+                  ],
+                ),
+              ),
+              const SizedBox(width: 10),
+              Expanded(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(L10n.of(context).endTime),
+                    const SizedBox(height: 10),
+                    TextFormField(
+                      readOnly: true,
+                      keyboardType: TextInputType.text,
+                      controller: _endTimeController,
+                      decoration: InputDecoration(
+                        hintText: L10n.of(context).selectTime,
+                        suffixIcon: const Icon(Icons.access_time_outlined),
+                      ),
+                      onTap: () => _selectTime(isStartTime: false),
+                      validator: (value) {
+                        if (value != null && value.isEmpty) {
+                          return L10n.of(context).endTimeRequired;
+                        }
+                        return null;
+                      },
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  // Selecting date
+  Future<void> _selectDate({required bool isStartDate}) async {
+    final date = await showDatePicker(
+      context: context,
+      initialDate: isStartDate ? _selectedStartDate : _selectedEndDate,
+      firstDate: DateTime.now(),
+      lastDate: DateTime.now().addYears(1),
+    );
+    if (date == null || !mounted) return;
+    if (isStartDate) {
+      _selectedStartDate = date;
+      _startDateController.text = eventDateFormat(date);
+      // if end date is empty and if start date is same or after end date
+      if (_endDateController.text.isEmpty ||
+          date.isSameOrAfter(_selectedEndDate)) {
+        _selectedEndDate = date;
+        _endDateController.text = eventDateFormat(date);
+      }
+    } else {
+      // if date is same or after start date
+      if (date.isSameOrAfter(_selectedStartDate)) {
+        _selectedEndDate = date;
+        _endDateController.text = eventDateFormat(date);
+        // When user change date that time end time is reset
+        _endTimeController.text = '';
+      } else {
+        EasyLoading.showToast(L10n.of(context).pleaseSelectValidEndDate);
+      }
+    }
+    setState(() {});
+  }
```

Comment:
nitpick: this sort of things screams for widget unittests to make sure they always work correctly and don't get broken upon later changes...